src/main/java/com/intuit/graphql/orchestrator/utils/QueryOptimizerUtil.java

```diff
@@ -0,0 +1,63 @@
+package com.intuit.graphql.orchestrator.utils;
+
+import graphql.language.Field;
+import graphql.language.SelectionSet;
+import org.apache.commons.lang3.ObjectUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class QueryOptimizerUtil {
```

Comment:
we should move the other code also here and have it as an instantialble class rather than util, Everything to do with query optimization can stay in this class.